build-common/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+use cbindgen::{self, Config};
+use std::path::{Path, PathBuf};
+use std::{env, fs};
+
+/// Generates a C header file using `cbindgen` for the specified crate.
+///
+/// # Arguments
+///
+/// * `crate_dir` - The directory of the crate to generate bindings for.
+/// * `header_name` - The name of the header file to generate.
+/// * `output_base_dir` - The base directory where the header file will be placed.
+///                       Defaults to target if unspecified
+pub fn generate_header(crate_dir: PathBuf, header_name: &str, output_base_dir: Option<&str>) {
+    let cargo_target_dir = output_base_dir.unwrap_or("target");
+
+    // Determine if `cargo_target_dir` is absolute or relative
+    let cargo_target_path = Path::new(cargo_target_dir);
+    let output_path = if cargo_target_path.is_absolute() {
+        // If absolute, use it directly
+        cargo_target_path.join("include/datadog/").join(header_name)
+    } else {
+        // If relative, adjust the path accordingly. we are in a crate, so get back to top level
+        let adjusted_path = Path::new("..").join(cargo_target_path);
```

Comment:
is this guaranteed to work? What if the command is run deeper in the crate?

Reply:
No.
However I can not think of a way to magically infer where the headers should be placed.
I looked into using the OUT_DIR variable, but that also has different depths depending on build parameters.
So I think this being not perfect still guarantees that it works in the use cases we have today.
Happy to reconsider if you find a different logic that works well.

Reply:
If we want the root of the workspace, can we try https://doc.rust-lang.org/cargo/commands/cargo-locate-project.html#options

Reply:
Here is a new iteration using your proposal.